include/zephyr/drivers/spi.h

```diff
@@ -891,8 +1026,17 @@ static inline int spi_write_dt(const struct spi_dt_spec *spec,
 {
 	return spi_write(spec->bus, &spec->config, tx_bufs);
 }
+/** @} */
 
 #if defined(CONFIG_SPI_ASYNC) || defined(__DOXYGEN__)
+/**
+ * @name SPI Asynchronous Transfer Functions
+ *
+ * With this API the transfer function will return immediately and
```

Comment:
This just isn't true as I've noted, it will return as soon as the transceive *starts* and that's not necessarily immediately.